pkg/virt-controller/services/template.go

```diff
@@ -1466,6 +1466,10 @@ func (t *templateService) renderLaunchManifest(vmi *v1.VirtualMachineInstance, i
 	} else if istio.ProxyInjectionEnabled(vmi) {
 		automount := true
 		pod.Spec.AutomountServiceAccountToken = &automount
+	} else if vmi.Spec.ServiceAccountName != "" {
+		pod.Spec.ServiceAccountName = vmi.Spec.ServiceAccountName
+		automount := true
+		pod.Spec.AutomountServiceAccountToken = &automount
```

Comment:
There's already a service account volume as well which sets the pod.Spec.ServiceAccountName. 

```
        // ServiceAccountVolumeSource represents a reference to a service account.
        // There can only be one volume of this type!
        // More info: https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/
        // +optional
        ServiceAccount *ServiceAccountVolumeSource `json:"serviceAccount,omitempty"`
```

How do we untangle this serviceAccount volume in addition to this new serviceAccount feature?

Reply:
The only feasible way is to match these two and reject otherwise. I will explore a possibility to support propagating another (less-privileged) SA as volume.

Reply:
Would it be possible to use the sa on the `vmi.Spec.ServiceAccountName` to create the pod but not set the service account on the pod spec?

For example, if we built a new client within virt-controller that uses the SA to authenticate when posting the pod?

Reply:
@davidvossel 
I am not sure how we would deliver this SA into the pod for the client(Well we would need to adhoc download the secret for it). In theory, we should be able to initialize the client from it. 

But I never tried this and afaik there is no out-of-box function where you would pass the path to the token...

Reply:
What I think we can re-implement is service account volume. We can just mount a secret associated with desired SA and propagate it to the VM. What do you think?

Reply:
oh yeah, that could make more sense actually if it is that simple

Reply:
@davidvossel Would you be in favor of merging this(only SA part) without solving the SA-SAvolume friction? I am not sure it is straightforward and there is also a bug around SA volume and migrations.